[Phua Lock Hian] iP

Phua-Lock-Hian opened the pull request.

Phua-Lock-Hian committed:
Increment: Level-0. Rename, Greet, Exit
Changed chatbot name to Swing

Phua-Lock-Hian committed:
Level 1 Echo added

Phua-Lock-Hian committed:
Level 2 List added

Phua-Lock-Hian committed:
Level 3 Tasks, Marking and Unmarking added. Also added customized responses.

Phua-Lock-Hian committed:
Level 4. ToDos, Events, Deadlines
added support for events and deadlines. to further refine and refactor.

samstt commented on src/main/java/Echo.java:
```diff
@@ -0,0 +1,18 @@
+import java.util.Scanner;
```
this appears to be a simple echo function, is creating an echo class really necessary?

Phua-Lock-Hian replied:
it was kept from one of the earlier increments for the iP, I wasn't sure if we were to keep it for grading purposes. but yes, you are right, functionally it is now not needed. thanks!

samstt commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+public class Event extends Task {
```
i like this, it is very clear and concise

samstt commented on src/main/java/List.java:
```diff
@@ -0,0 +1,109 @@
+import java.util.Scanner;
+
+public class List {
```
the logical flow of this code was very clear! good job!

samstt commented on src/main/java/List.java:
```diff
@@ -0,0 +1,109 @@
+        return false;
+    }
+
+    private void errorMessage() {
+        System.out.println("uh oh, something went wrong meow. try again~");
+        System.out.println("--");
+    }
```
good job catching the errors in your chatbot

samstt commented on src/main/java/List.java:
```diff
@@ -0,0 +1,109 @@
+        Scanner in = new Scanner(System.in);
+        do {
+            System.out.println("--");
+            line = in.nextLine().trim();
```
since java is case sensitive, what if the user inputs 'Bye' or 'List'? will your chatbot still be able to run?

Phua-Lock-Hian replied:
good catch! i'll fix that.

samstt commented on src/main/java/Swing.java:
```diff
@@ -0,0 +1,12 @@
+//import java.util.Scanner;
+
+public class Swing {
+    public static void main(String[] args) {
```
main function is very concise. good!

samstt reviewed (commented):
Overall, i really like the way your code is presented and i didn't really notice any violations in coding standards. Good job!

vincesum commented on src/main/java/List.java:
```diff
@@ -0,0 +1,109 @@
+    private int len;
+
+    public List() {
+        list = new Task[100];
```
As it is, 100 is a magic number. Have you considered changing the 100 to a constant?

Phua-Lock-Hian replied:
good catch! i'll fix that.

vincesum commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+        this.end = end;
+    }
+
+    @Override
```
Nice override annotation, very clear

vincesum commented on src/main/java/List.java:
```diff
@@ -0,0 +1,109 @@
+            String[] parts = line.split(" ", 2);
+            String command = parts[0];
+            System.out.println("--");
+            switch (command) {
```
It might be better to implement SLAP by refactoring the code below each case into different methods.

Phua-Lock-Hian replied:
agreed, was planning to refactor. thanks!

vincesum commented on src/main/java/List.java:
```diff
@@ -0,0 +1,109 @@
+                System.out.println(list[toUnmark - 1].toString());
+                break;
+
+            case "deadline":
```
You might want to test whether deadline entered was valid by checking for /by in the second part

Phua-Lock-Hian replied:
good catch, thanks

vincesum commented on src/main/java/List.java:
```diff
@@ -0,0 +1,109 @@
+            case "event":
+                if (!isValidCommand(parts, 2)) break;
+
+                String[] eventArgParts = parts[1].split("/", 3);
```
Have you considered what happens when the person does not follow the format of entering /from and /to and just entered // instead?

Phua-Lock-Hian replied:
good catch, that's a bug. i'll fix it.

vincesum commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
```
Good usage of inheritance to reduce code needed.

vincesum commented on src/main/java/List.java:
```diff
@@ -0,0 +1,109 @@
+        System.out.println("Now you have " + len + " tasks in list.");
+    }
+
+    private boolean isValidCommand(String[] parts, int expectedLen) {
```
Well done on checking whether command entered is valid!

vincesum reviewed (commented):
Overall, well done on your chatbot Swing! There are some conventions that can be better followed and a few bugs but it generally should work well!

Phua-Lock-Hian committed:
edits according to PRs, bug fixes and code quality improvements

Phua-Lock-Hian committed:
Level 5 added, handling errors using exceptions.

Phua-Lock-Hian committed:
Merge branch 'branch-Level-5'

Phua-Lock-Hian committed:
code refactor into packages

Phua-Lock-Hian committed:
Merge branch 'branch-A-Packages'

Phua-Lock-Hian committed:
Level 6 added, deleting now supported.
Bug fixes.
Now using ArrayList instead of static array.

Phua-Lock-Hian committed:
Level 7 work in progress

Phua-Lock-Hian committed:
Level 7 added, saving now supported. Data is stored in data.txt

Phua-Lock-Hian committed:
Completed Level 6 and Level 7 of iP (delete, save)

Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/swing/List.java

Phua-Lock-Hian committed:
JAR file created

Phua-Lock-Hian committed:
Bug fixes for saving data.

Phua-Lock-Hian committed:
Refactoring code to meet A-MoreOOP requirements. Switched to using regex to load data from file to support wider range of inputs (original splitstr approach only supports single word descriptions)

Phua-Lock-Hian committed:
Level-9 Find added. The search supports keywords within to/from/by params as well.

Phua-Lock-Hian committed:
Added JavaDoc comments to public classes and methods

Phua-Lock-Hian committed:
Merge pull request #1 from Phua-Lock-Hian/branch-Level-9

Level-9 Find

Phua-Lock-Hian committed:
Merge branch 'master' into branch-A-JavaDoc

Phua-Lock-Hian committed:
Merge pull request #2 from Phua-Lock-Hian/branch-A-JavaDoc

Added JavaDoc comments to public classes and methods

Phua-Lock-Hian committed:
Merge branch 'master' of https://github.com/Phua-Lock-Hian/ip

Phua-Lock-Hian committed:
Bug fix in Parser. Added User Guide.

wenxin-c commented on src/main/java/swing/Parser.java:
```diff
@@ -0,0 +1,204 @@
+package swing;
+
+import swing.tasktypes.*;
```
Please import libraries/classes explicitly instead of using wildcard.

wenxin-c commented on src/main/java/swing/Parser.java:
```diff
@@ -0,0 +1,204 @@
+        return false;
+    }
+
+    private void errorMessage(Ui ui) {
+        ui.showMessage("uh oh, something went wrong meow. try again~");
+    }
```
You could consider naming the method sound more verb, e.g. ```printErrorMessage```.

wenxin-c commented on src/main/java/swing/Parser.java:
```diff
@@ -0,0 +1,204 @@
+        ui.showMessage("uh oh, something went wrong meow. try again~");
+    }
+
+    private void addedToList(TaskList tasks, Ui ui) {
+        ui.showMessage("added: " + tasks.get(tasks.size() - 1).toString()); //to offset zero indexing
+        ui.showMessage("Now you have " + tasks.size() + " tasks in list.");
+    }
```
Similarly here, a more verb name would be better, e.g. ```addToList```.

wenxin-c commented on src/main/java/swing/Storage.java:
```diff
@@ -0,0 +1,114 @@
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import swing.tasktypes.*;
```
Refer to the previous comment on wildcard.

wenxin-c commented on src/main/java/swing/Storage.java:
```diff
@@ -0,0 +1,114 @@
+    public static void createFile(){
+        try {
+            // Create file to store data if it does not exist
+            File f = new File("./data.txt");
```
You might consider abstracting the file path into a named constant, so that since it's used multiple times, it will be easier for future changes.

wenxin-c commented on src/main/java/swing/Storage.java:
```diff
@@ -0,0 +1,114 @@
+            while (scan.hasNextLine()) {
+                String line = scan.nextLine();
+                Matcher matcher;
+                if (line.startsWith("[T]")) {
```
You might consider extracting the command type and using ```switch``` to check for commands instead of calling ```startsWith``` multiple times.

wenxin-c commented on src/main/java/swing/Storage.java:
```diff
@@ -0,0 +1,114 @@
+                        System.out.println("Error parsing deadline: " + line);
+                    }
+                } else if (line.startsWith("[E]")) {
+                    matcher = eventPattern.matcher(line);
+                    if (matcher.matches()) {
+                        boolean isDone = matcher.group(1).trim().equals("X");
+                        String desc = matcher.group(2);
+                        String from = matcher.group(3);
+                        String to = matcher.group(4);
+                        Event newEvent = new Event(desc, from, to);
+                        newEvent.setStatusIcon(isDone);
+                        taskList.add(newEvent);
```
You might consider abstracting each block into a separate method so that code will be neater and more organised.

wenxin-c reviewed (commented):
Good job in your IP. There are some comments for your consideration. You could consider abstracting more to make code neater and more organised.

Phua-Lock-Hian committed:
Edits to code style based on TA feedback

Phua-Lock-Hian committed:
Edits to code style based on TA feedback

Phua-Lock-Hian committed:
Merge remote-tracking branch 'origin/master'

Phua-Lock-Hian committed:
editing comments

Phua-Lock-Hian committed:
bug fixes after user testing. fixed:
(1) deadline: crashes when there is whitespace " " entered after /by
(2) event: crashes when there is nothing after /to or " " after /to
(3) text file saved after the user exits the program, when it shld be after every command

Phua-Lock-Hian committed:
fixed bug where program crashes if there is no input after "find"
